internal/cmd/app.go

```diff
@@ -62,7 +62,15 @@ func NewApp(o *globals.GlobalOpts) *cli.App {
 			DefaultText: globals.DefaultEnvoyVersionsURL,
 			Destination: &envoyVersionsURL,
 			EnvVars:     []string{"ENVOY_VERSIONS_URL"},
-		}}
+		},
+		&cli.StringFlag{
+			Name:        "platform",
+			Usage:       "target platform to install Envoy",
+			DefaultText: globals.DefaultPlatform,
+			Destination: &envoyVersionsURL,
```

Comment:
```suggestion
			Destination: &envoyPlatform,
```

then later you would check if "" and set o.Platform accordingly

Reply:
@codefromthecrypt  The  TestInstallIfNeeded is getting the incorrect date from `string(test.FakeReleaseDate)`
```
internal/envoy - (platform-flag) $ go test ./...
--- FAIL: TestInstallIfNeeded (0.09s)
    install_test.go:168:
        	Error Trace:	install_test.go:168
        	Error:      	Not equal:
        	            	expected: "2020-12-30"
        	            	actual  : "2020-12-31"

        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-2020-12-30
        	            	+2020-12-31
        	Test:       	TestInstallIfNeeded
FAIL
FAIL	github.com/tetratelabs/func-e/internal/envoy	2.264s
```
The test snippet failing.
```go
	f, err := os.Stat(versionDir)
	require.NoError(t, err)
	require.Equal(t, f.ModTime().Format("2006-01-02"), string(test.FakeReleaseDate))
```
 
Should I change `FakeReleaseDate = version.ReleaseDate("2020-12-31")` ?

Reply:
hmm maybe insert `.UTC()` between ModTime and Format?

Reply:
if this doesn't work, I'd suggest to focus on the other tests and push whatever you can. I can take a look and help on this one!

Reply:
@codefromthecrypt Yeah, I'm seeing various failed test due to time offset by 1. I'm wrapping up the test for the platform arg and fixing the Usage.MD test.  I'll have a push here in a bit that I'll tag you into review/

Reply:
@codefromthecrypt Okay, I think I'm all done on my end. If you have a moment, can you check out the few checks that are failing due to the time offset please ?

Reply:
Yeap sure will and try to help you land this so https://github.com/tetratelabs/func-e/pull/301 doesn't drift it!